Atomic/mount.py

```diff
@@ -87,12 +88,9 @@ def set_args(self, args):
 
     def mount(self):
         try:
-            try:
-                d = OSTreeMount(self.mountpoint, live=self.live, shared=self.shared)
-                if d.mount(self.image, self.options):
-                    return
-            except (MountError,NoDockerDaemon):
-                pass
+            d = OSTreeMount(self.mountpoint, live=self.live, shared=self.shared)
+            if d.mount(self.image, self.options):
+                return
 
```

Comment:
yes, I moved the failing exception after we check that there is a system container with that name.  It is needed so I can properly show an error message when users specify --live or --shared with a system container.

To avoid reintroducing the issue, I have added some more tests for --live and --shared when used on a regular  Docker container.